src/platform/forms-system/src/js/patterns/array-builder/arrayBuilder.jsx

```diff
@@ -336,63 +336,96 @@ export function arrayBuilderPages(options, pageBuilderCallback) {
   };
 
   // Verify and setup any initial page options
-  const testConfig = pageBuilderCallback(pageBuilderVerifyAndSetup);
+  pageBuilderCallback(pageBuilderVerifyAndSetup);
   validatePages(orderedPageTypes);
   validateRequired(userRequired);
   validateReviewPath(reviewPath);
   validateMinItems(options.minItems);
   const required =
     typeof userRequired === 'function' ? userRequired : () => userRequired;
-  const pageKeys = Object.keys(testConfig);
-  const firstItemPagePath = itemPages?.[0]?.path;
   const lastItemPagePath = itemPages?.[itemPages.length - 1]?.path;
-  const itemLastPageKey = pageKeys?.[pageKeys.length - 1];
+
+  const getActiveItemPages = (formData, index) => {
+    return itemPages.filter(page => {
+      try {
+        if (page.depends) {
+          return typeof page.depends === 'function'
+            ? page.depends(formData, index)
+            : page.depends;
+        }
+        return true;
+      } catch (e) {
+        return false;
+      }
+    });
+  };
+
+  const getFirstItemPagePath = (formData, index) => {
+    return getActiveItemPages(formData, index)?.[0]?.path;
+  };
+
+  const getLastItemPagePath = (formData, index) => {
+    const activeItemPages = getActiveItemPages(formData, index);
+    return activeItemPages?.[activeItemPages.length - 1]?.path;
+  };
 
   // Didn't throw error so success: Validated and setup success
   const pageBuilder = pageBuilderVerifyAndSetup;
 
   /** @type {FormConfigPage['onNavForward']} */
-  const navForwardFinishedItem = ({ goPath, urlParams, pathname }) => {
+  const navForwardFinishedItem = ({ goPath, urlParams, pathname, index }) => {
     let path = summaryPath;
     if (urlParams?.edit || (urlParams?.add && urlParams?.review)) {
-      const index = getArrayIndexFromPathName(pathname);
+      const foundIndex = getArrayIndexFromPathName(pathname);
       const basePath = urlParams?.review ? reviewPath : summaryPath;
       path = createArrayBuilderUpdatedPath({
         basePath,
-        index,
+        index: foundIndex == null ? index : foundIndex,
         nounSingular,
       });
     }
     goPath(path);
   };
 
+  const navForwardSummaryAddItem = ({ formData, goPath }) => {
+    const nextIndex = formData[arrayPath]?.length || 0;
+
+    const path = createArrayBuilderItemAddPath({
+      path: getFirstItemPagePath(formData, nextIndex),
+      index: nextIndex,
+    });
+    goPath(path);
+  };
+
+  const navForwardSummaryGoNextChapter = ({ formData, goPath, pageList }) => {
```

Comment:
This part is new in the latest changes, which fixes the bugs that @williamphelps13 had discovered. (The index was sometimes wrong in a previous commit for this PR)